feathr_project/feathr/definition/sink.py

```diff
@@ -100,4 +112,61 @@ def to_feature_config(self) -> str:
             }
         """)
         hocon_config = tm.render(sink=self)
-        return hocon_config
\ No newline at end of file
+        return hocon_config
+
+    def to_argument(self):
+        return self.output_path
+
+    
+class GenericSink(Sink):
+    def __init__(self, format: str, mode: Optional[str] = None, options: Dict[str, str] = {}) -> None:
+        self.format = format
+        self.mode = mode
+        self.options = dict([(o.replace(".", "__"), options[o]) for o in options])
+    
+    def to_feature_config(self) -> str:
+        ret = {
+            "name": "HDFS",
+            "params": self._to_dict()
+        }
+        return json.dumps(ret, indent=4)
+    
+    def _to_dict(self) -> dict[str, str]:
+        ret = self.options.copy()
+        ret["type"] = "generic"
+        ret["format"] = self.format
+        if self.mode:
+            ret["mode"] = self.mode
+        return ret        
+    
+    def get_required_properties(self):
+        ret = []
+        for option in self.options:
+            start = option.find("${")
+            if start >= 0:
+                end = option[start:].find("}")
+                if end >= 0:
+                    ret.append(option[start+2:start+end])
+        return ret
+
+    def to_argument(self):
+        """
+        One-line JSON string, used by job submitter
+        """
+        return json.dumps(self._to_dict())
+    
+class CosmosDbSink(GenericSink):
+    def __init__(self, name: str, endpoint: str, database: str, collection: str):        
```

Comment:
should change collection to container to reduce confusion

Reply:
Changed